source/common/router/config_impl.h

```diff
@@ -194,6 +194,12 @@ class VirtualHostImpl : public VirtualHost {
 private:
   enum class SslRequirements { None, ExternalOnly, All };
 
+  struct StatNameProvider {
+    StatNameProvider(absl::string_view name, Stats::SymbolTable& symbol_table)
+        : stat_name_storage_(std::make_unique<Stats::StatNameManagedStorage>(name, symbol_table)) {}
+    std::unique_ptr<Stats::StatNameManagedStorage> stat_name_storage_;
```

Comment:
Is there any reason this needs to be heap allocated within the struct? Can it be inline?

Reply:
I thought there was some order-of-init issue but evidently not. Done.